### PR TITLE
Make schema robust

### PR DESCRIPTION
**Pull request by j-zimnowoda**
#490

### CODE REVIEW

File: values-schema.yaml [hunk 1]
```diff
@@ -138,17 +138,17 @@ definitions:
       region:
         title: AWS region
         description: An AWS region.
-        type: string
+        $ref: '#/definitions/wordCharacterPattern'
```
Comment (Morriz) [inline values-schema.yaml:141]: why not add the enum of AWS here?

File: values-schema.yaml [hunk 2]
```diff
@@ -138,17 +138,17 @@ definitions:
...
         example: arn:aws:iam::YYYYYYYYYYYY:role/dns-manager
-        type: string
+        $ref: '#/definitions/wordCharacterPattern'
   azure:
     definitions:
       clientId:
         title: Azure client id
         description: An Azure client id.
-        type: string
+        $ref: '#/definitions/wordCharacterPattern'
```
Comment (Morriz) [inline values-schema.yaml:151]: maybe same with client id which always looks the same?

File: values-schema.yaml (hunk 1 above)
Comment (Morriz) [reply values-schema.yaml:141]: And maybe add all the enums for all the cloud providers, and then choose a `oneOf` for `cluster.region`

**Review by Morriz — CHANGES_REQUESTED** [review]
good initial improvement...could definitely use much more love...how about the following:
- enums where we can
- enforce combinations of properties that are required together (I leave it up to the one doing the spike on this one what is now allowed and breaking our install...go ahead and be thoroughly creative ;)

File: values-schema.yaml [hunk 3]
```diff
@@ -588,11 +592,24 @@ definitions:
         $ref: '#/definitions/resource'
         description: Requested resources (guaranteed).
     title: Pod resources
-  runAsUser:
+  linuxUserIdRange:
+    type: object
```
Comment (svatwork) [inline values-schema.yaml:596]:
```suggestion
```

**Review by svatwork — CHANGES_REQUESTED** [review]
Schema is more robust now...? I don't understand how this improves things. We need a structural improvement, no? Otherwise we'll have regression again.

File: values-schema.yaml (hunk 3 above)
Comment (Dunky13) [reply values-schema.yaml:596]: `type` must be explicit, due to Ajv (validate script). Otherwise it will compain.

File: values-schema.yaml [hunk 4]
```diff
@@ -27,17 +27,17 @@ definitions:
       groupInterval:
         default: 5m
         description: How long to wait before sending a notification about new alerts that are added to a group of alerts for which an initial notification has already been sent. (Usually ~5m or more.)
-        type: string
+        $ref: '#/definitions/wordCharacterPattern'
       msteams:
         additionalProperties: false
         properties:
           highPrio:
             description: The low prio web hook.
```
Comment (Dunky13) [inline values-schema.yaml:35]: Shouldn't this be `The high prio web hook.`?

File: values-schema.yaml [hunk 5]
```diff
@@ -27,17 +27,17 @@ definitions:
...
-            type: string
+            $ref: '#/definitions/wordCharacterPattern'
             x-secret: ''
           lowPrio:
             description: The high prio web hook.
```
Comment (Dunky13) [inline values-schema.yaml:39]: Shouldn't this be `The low prio web hook.`?

File: values-schema.yaml [hunk 6]
```diff
@@ -27,17 +27,17 @@ definitions:
       groupInterval:
         default: 5m
         description: How long to wait before sending a notification about new alerts that are added to a group of alerts for which an initial notification has already been sent. (Usually ~5m or more.)
-        type: string
+        $ref: '#/definitions/wordCharacterPattern'
```
Comment (Dunky13) [inline values-schema.yaml:30]: Should we not make a `timePattern` that maps `<digit><timeunit>` so something like: `^[0-9]+[smh]$` for example?

File: values-schema.yaml [hunk 7]
```diff
@@ -51,18 +51,18 @@ definitions:
       repeatInterval:
         default: 3h
         description: How long to wait before sending a notification again if it has already been sent successfully for an alert. (Usually ~3h or more).
-        type: string
+        $ref: '#/definitions/wordCharacterPattern'
```
Comment (Dunky13) [inline values-schema.yaml:54]: Time pattern mentioned earlier

**Review by Dunky13 — COMMENTED** [review]
A couple of minor suggestions

**Review by mojtabaimani — APPROVED** [review]
Looks good to me. I suggest other improvements goes to another PR.

**Comment by Morriz** [comment]
If we are to do another PR, then we must be sure to put all the observations from here as requirements there. Seems like overhead to me, but fine by me. Who will make a new PR that holds all of that?

**Comment by j-zimnowoda** [comment]
For all other findings, I am going to create a new story that precisely describes all required changes.

**Review by Dunky13 — APPROVED** [review]
LGTM

File: values-schema.yaml [hunk 8]
```diff
@@ -487,9 +493,9 @@ definitions:
       - properties:
           fsGroup:
             description: Supplementary group ID. Volumes that support ownership management are modified to be owned and writable by this ID.
-            type: integer
+            $ref: '#/definitions/linuxUserId'
```
Comment (Morriz) [inline values-schema.yaml:496]: why not `unixOwnerId`? I don't think it has something to do with linux specifically, and I see it being used for groups as well

**Review by Morriz — CHANGES_REQUESTED** [review]
one naming issue left

File: values-schema.yaml (hunk 8 above)
Comment (j-zimnowoda) [reply values-schema.yaml:496]: I think most of our nodes are Linux based

Comment (Morriz) [reply values-schema.yaml:496]: well, we are referring to unix constructs, not linux..

**Pull request merged**